packages/nocodb/Dockerfile

```diff
@@ -18,25 +18,30 @@ RUN cp $GOPATH/bin/litestream /usr/src/lt
 ###########
 # Builder
 ###########
-FROM node:16.17.0-alpine3.15 as builder
+FROM node:18.12.1-alpine as builder
 WORKDIR /usr/src/app
 
 # install node-gyp dependencies
 RUN apk add --no-cache python3 make g++
 
+# install pnpm
+RUN corepack enable && corepack prepare pnpm@latest --activate
+
 # Copy application dependency manifests to the container image.
-# A wildcard is used to ensure both package.json AND package-lock.json are copied.
-# Copying this separately prevents re-running npm ci on every code change.
-COPY ./package*.json ./
+COPY ./package.json ./package.json
 COPY ./docker/main.js ./docker/main.js
 #COPY ./docker/start.sh /usr/src/appEntry/start.sh
 COPY ./docker/start-litestream.sh /usr/src/appEntry/start.sh
 COPY src/public/ ./docker/public/
 
+# for pnpm to generate a flat node_modules without symlinks
+# so that modclean could work as expected
+RUN echo "node-linker=hoisted" > .npmrc
+
 # install production dependencies,
 # reduce node_module size with modclean & removing sqlite deps,
 # package built code into app.tar.gz & add execute permission to start.sh
-RUN npm ci --omit=dev --quiet \
+RUN pnpm install --prod --shamefully-hoist \
     && npx modclean --patterns="default:*" --ignore="nc-lib-gui/**,dayjs/**,express-status-monitor/**,@azure/msal-node/dist/**" --run  \
```

Comment:
Can we use `pnpm exec modclean` here ?

Reply:
Good catch. However, I think it should be `pnpm dlx` instead since `modclean` is not in our dependencies.